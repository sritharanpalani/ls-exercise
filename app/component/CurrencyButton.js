import React, {useState} from 'react';
import {
  View,
  StyleSheet,
  Text,
  TouchableOpacity,
  SafeAreaView,
  Dimensions,
} from 'react-native';

const width = Dimensions.get('window').width;

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    margin: 16,
  },
  tabContainer: {
    display: 'flex',
    width: width - 32,
  },
  buttonContainer: {
    height: 32,
    width: 100,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: '#000341',
    margin: 8,
  },
  buttonTitle: {
    fontSize: 20,
    alignSelf: 'center',
  },
  divider: {
    marginTop: 16,
    width: width,
    height: 1,
    backgroundColor: '#d4d4d4',
  },
});

const CurrencyButton = props => {
  const [selectedIndex, setSelectedIndex] = useState(null);

  const currenyButtonItems = [
    {id: 0, name: 'INR'},
    {id: 1, name: 'AED'},
    {id: 2, name: 'SAR'},
  ];

  const handleTap = (index, item) => {
    const {conversionCalculation} = props;
    setSelectedIndex(index);
    conversionCalculation(index, item);
  };

  return (
    <SafeAreaView>
      <View style={styles.container}>
        {currenyButtonItems.map(item => (
          <TouchableOpacity
            key={item.id}
            onPress={() => handleTap(item.id, item)}
            style={[
              styles.buttonContainer,
              {
                backgroundColor:
                  selectedIndex === item.id ? '#f4f699' : 'transparent',
              },
            ]}>
            <View>
              <Text style={styles.buttonTitle}>{item.name}</Text>
            </View>
          </TouchableOpacity>
        ))}
      </View>
      <View style={styles.divider} />
    </SafeAreaView>
  );
};

export default CurrencyButton;
