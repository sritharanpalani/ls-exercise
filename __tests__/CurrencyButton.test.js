import * as React from 'react';
import renderer from 'react-test-renderer';

import CurrencyButton from '../app/component/CurrencyButton';

test('<CurrencyButton> renders correctly', () => {
  const tree = renderer.create(<CurrencyButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
