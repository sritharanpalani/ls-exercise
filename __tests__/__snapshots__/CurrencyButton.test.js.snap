// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyButton> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 16,
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#000341",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 32,
          "margin": 8,
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 20,
            }
          }
        >
          INR
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#000341",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 32,
          "margin": 8,
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 20,
            }
          }
        >
          AED
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#000341",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 32,
          "margin": 8,
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 20,
            }
          }
        >
          SAR
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#d4d4d4",
        "height": 1,
        "marginTop": 16,
        "width": 750,
      }
    }
  />
</RCTSafeAreaView>
`;
